pkg/inference/backends/llamacpp/download_linux.go

```diff
@@ -1,18 +1,50 @@
+//go:build linux
+
 package llamacpp
 
 import (
 	"context"
 	"fmt"
 	"net/http"
-	"path/filepath"
 
+	"github.com/docker/model-runner/pkg/gpuinfo"
 	"github.com/docker/model-runner/pkg/logging"
 )
 
-func (l *llamaCpp) ensureLatestLlamaCpp(_ context.Context, log logging.Logger, _ *http.Client,
-	_, vendoredServerStoragePath string,
+func init() {
+	// Enable GPU variant detection by default on Linux
+	ShouldUseGPUVariantLock.Lock()
+	defer ShouldUseGPUVariantLock.Unlock()
+	ShouldUseGPUVariant = true
+}
```

Comment:
I'd rather have `ShouldUseGPUVariant       = runtime.GOOS == "linux"` in `download.go` instead of an `init`.
For more context, this variable is used by Docker Desktop to determine whether it should display the GPU backed engine checkbox in the UI.